crates/holochain_sqlite/src/db/p2p_agent_store.rs

```diff
@@ -156,34 +157,55 @@ impl AsP2pStateTxExt for Transaction<'_> {
         Ok(out)
     }
 
-    fn p2p_gossip_query(
+    fn p2p_gossip_query_agents(
         &self,
         since_ms: u64,
         until_ms: u64,
-        within_arc: DhtArc,
-    ) -> DatabaseResult<Vec<KitsuneAgent>> {
+        arcset: DhtArcSet,
+    ) -> DatabaseResult<Vec<(KitsuneAgent, ArcInterval)>> {
         let mut stmt = self
             .prepare(sql_p2p_agent_store::GOSSIP_QUERY)
             .map_err(|e| rusqlite::Error::ToSqlConversionFailure(e.into()))?;
 
-        let (start, end) = within_arc.primitive_range_transposed();
-
-        let mut out = Vec::new();
-        for r in stmt.query_map(
-            named_params! {
-                ":since_ms": clamp64(since_ms),
-                ":until_ms": clamp64(until_ms),
-                ":storage_start_loc": start,
-                ":storage_end_loc": end,
-            },
-            |r| {
-                let agent: Vec<u8> = r.get(0)?;
-                Ok(KitsuneAgent(agent))
-            },
-        )? {
-            out.push(r?);
-        }
-        Ok(out)
+        let query = stmt
+            .query_map(
+                named_params! {
+                    ":since_ms": clamp64(since_ms),
+                    ":until_ms": clamp64(until_ms),
+                    // we filter by arc in memory, not in the db query
+                    ":storage_start_loc": Some(u32::MIN),
+                    ":storage_end_loc": Some(u32::MAX),
```

Comment:
I think we may want a simpler query that doesn't compare these values on the sql side.